### PR TITLE
ASimRegistryEventMicrosoftSysmon.yaml-52

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for WindowsEvent and added WindowsEvent parser in it. In Event file removed WindowsEvent parser and in ASimRegistryEvent added WindowsEvent Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the WindowsEvent and Event parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please resolve KQL validation failures. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please act on above comment. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
We are waiting for your response. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_DataTest.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 10 records (100.0%) for field [EventVendor] of type [Enumerated]: [""Microsoft""] (Schema:RegistryEvent)"
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_DataTest.csv:2]: please check these errors

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv [hunk 2]
```diff
@@ -0,0 +1,45 @@
+﻿Result
+"(0) Error: Missing mandatory field [EventResult]"
+"(0) Error: Missing mandatory field [EventSchema]"
+"(0) Error: Missing mandatory field [ParentProcessId]"
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv:4]: please check these errors

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistry.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/Parsers/ASimRegistry.yaml:1]: Please rename this file to ASimRegistryEvent.yaml

File: Parsers/ASimRegistryEvent/Parsers/imRegistry.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/Parsers/imRegistry.yaml:1]: Please rename this file to imRegistryEvent.yaml

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer. Please act on them.
image
And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistry.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistry.yaml:1]: updating table schema file

Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistry.yaml:1]: done

File: Parsers/ASimRegistryEvent/Parsers/imRegistry.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/imRegistry.yaml:1]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check and responds on open comments. Thanks

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_AsimRegistryEvent_SchemaTest.csv [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿Result
+"(0) Error: Missing mandatory field [EventResult]"
+"(0) Error: Missing mandatory field [EventSchema]"
+"(0) Error: Missing mandatory field [ParentProcessId]"
+"(0) Error: Missing optional alias [Rule] aliasing existing column [RuleName]"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_AsimRegistryEvent_SchemaTest.csv]: parentprocessid haven't done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_SchemaTest.csv [hunk 6]
```diff
@@ -0,0 +1,44 @@
+﻿Result
+"(0) Error: Missing mandatory field [EventResult]"
+"(0) Error: Missing mandatory field [EventSchema]"
+"(0) Error: Missing mandatory field [ParentProcessId]"
+"(0) Error: Missing optional alias [Rule] aliasing existing column [RuleName]"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_SchemaTest.csv]: parentprocessid haven't done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_AsimRegistryEvent_SchemaTest.csv [hunk 7]
```diff
@@ -0,0 +1,45 @@
+﻿Result
+"(0) Error: Missing mandatory field [EventResult]"
+"(0) Error: Missing mandatory field [EventSchema]"
+"(0) Error: Missing mandatory field [ParentProcessId]"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_AsimRegistryEvent_SchemaTest.csv]: parentprocessid haven't done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv]: parentprocessid haven't done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_DataTest.csv (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_DataTest.csv:2]: Added supported type in tester file

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_DataTest.csv [hunk 8]
```diff
@@ -0,0 +1,5 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 10 records (100.0%) for field [EventVendor] of type [Enumerated]: [""Microsoft""] (Schema:RegistryEvent)"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_DataTest.csv:2]: updated supported values in tester file

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_AsimRegistryEvent_SchemaTest.csv (hunk 5 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_AsimRegistryEvent_SchemaTest.csv]: you need help with this Alekhya?

[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_AsimRegistryEvent_SchemaTest.csv]: done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_AsimRegistryEvent_SchemaTest.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_AsimRegistryEvent_SchemaTest.csv]: done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_Windows_Event_VimRegistryEvent_SchemaTest.csv]: done

File: Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_SchemaTest.csv (hunk 6 above)
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/test/Microsoft_Sysmon_VimRegistryEvent_SchemaTest.csv]: done

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEvent.yaml [hunk 9]
```diff
@@ -37,7 +38,8 @@ ParserQuery: |
     vimRegistryEventEmpty,
     ASimRegistryEventMicrosoft365D(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoft365D' in (DisabledParsers) ))),
     ASimRegistryEventMicrosoftSysmon(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoftSysmon' in (DisabledParsers) ))),
-    ASimRegistryEventMicrosoftWindowsEvent(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoftWindowsEvent' in (DisabledParsers) ))),
+    ASimRegistryEventMicrosoftSysmonWindowsEvent(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoftSysmonWindowsEvent' in (DisabledParsers) ))),
+    ASimRegistryEventMicrosoftWindowsEvent(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoftSysmonWindowsEvent' in (DisabledParsers) ))),
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/Parsers/ASimRegistryEvent.yaml:42]: ASimRegistryEventMicrosoftWindowsEvent(disabled=(ASimBuiltInDisabled or ('ExcludeASimRegistryEventMicrosoftWindowsEvent' in (DisabledParsers) )))

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml [hunk 10]
```diff
@@ -93,76 +93,16 @@ ParserQuery: |
...
                   RegistryKey = iff (EventType == "RegistryKeyRenamed", NewKey, Key),
                   RegistryKeyModified = iff (EventType in ("RegistryKeyRenamed", "RegistryValueSet"), Key, ""),
                   RegistryValue = iff (EventType in ("RegistryValueSet", "RegistryValueDeleted"), Value, ""),
                   RegistryValueModified = iff (EventType == "RegistryValueSet", Value, ""),
                   RegistryValueData = iff (EventType == "RegistryValueSet", Parameter, ""),
-                  ActorUsernameType =  iff(isnotempty(ActorUsername), 'Windows', '')
+                  EventResult = "Success",
+                  EventSchema = "RegistryEvent",
+                  Rule=RuleName,
+                  ParentProcessId=ActingProcessId
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml:105]: how we concluded this?

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml [hunk 11]
```diff
@@ -173,12 +113,11 @@ ParserQuery: |
                   Key,
                   NewKey,
                   NewValue,
-                  EventData
+                  EventData,
+                  ParameterXml,DvcHostName,EventCategory,EventID,EventLevelName,EventLevel,EventLog,Hive1,MG,AzureDeploymentID,RegistryKeyModified,RegistryValueModified,Role,SourceSystem,Source,TenantId,UserName,UtcTime,ManagementGroupName,Message,_ResourceId
```
Comment (vakohl) [inline Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml:117]: can we correct the formatting?

Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml:117]: done

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml:105]: checked in document,other parsers and tested in workspace

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEvent.yaml (hunk 9 above)
Comment (Alekhya0824) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistryEvent.yaml:42]: added

File: Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml (hunk 10 above)
Comment (vakohl) [reply Parsers/ASimRegistryEvent/Parsers/ASimRegistryEventMicrosoftSysmon.yaml]: didn't map ParentProcessId as value not found in source

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**